dhall/src/Dhall/Import.hs

```diff
@@ -746,6 +749,36 @@ toHeader _ = do
 getCacheFile
     :: (Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
 getCacheFile cacheName hash = do
+    cacheDirectory <- getOrCreateCacheDirectory cacheName
+
+    let cacheFile = cacheDirectory </> ("1220" <> show hash)
+
+    return cacheFile
+
+warnAboutMissingCaches  :: (Alternative m, MonadIO m) => m ()
+warnAboutMissingCaches = do
+    warnings <- traverse warnAboutMissingCache ["dhall", "dhall-haskell"]
+
+    case catMaybes warnings of
+        warn:_ -> liftIO (System.IO.hPutStrLn System.IO.stderr warn)
+        []     -> return () 
+      
+
+warnAboutMissingCache :: (Alternative m, MonadIO m) => FilePath -> m (Maybe String)
+warnAboutMissingCache cacheName = alternative₀ <|> alternative₁
+  where
+    
+    alternative₀ = getOrCreateCacheDirectory cacheName >> return Nothing
+
+    alternative₁ = return (Just warning)
+
+      where warning = "It has not been possible to get a cache directory \"" ++ cacheName ++ "\" with read/write permission.\n"
```

Comment:
I think it would be better if we would report the full patch of the cache directory here.

Reply:
Agree, but i will have to change the logic to make available the possible result of `getOrCreateCacheDirectory`. 
I'll give a try...